src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2357,3 +2364,1537 @@ IsForeignTable(Oid relationId)
+		}
+
+		case BACKGROUND_JOB_STATUS_RUNNING:
+		{
+			return CitusJobStatusRunningId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FINISHED:
+		{
+			return CitusJobStatusFinishedId();
+		}
+
+		case BACKGROUND_JOB_STATUS_CANCELLED:
+		{
+			return CitusJobStatusCancelledId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FAILING:
+		{
+			return CitusJobStatusFailingId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FAILED:
+		{
+			return CitusJobStatusFailedId();
+		}
+	}
+
+	elog(ERROR, "unknown BackgroundJobStatus");
+	return InvalidOid;
+}
+
+
+/*
+ * BackgroundTaskStatusOid returns the Oid corresponding to SQL enum value corresponding to
+ * the BackgroundTaskStatus.
+ */
+Oid
+BackgroundTaskStatusOid(BackgroundTaskStatus status)
+{
+	switch (status)
+	{
+		case BACKGROUND_TASK_STATUS_BLOCKED:
+		{
+			return CitusTaskStatusBlockedId();
+		}
+
+		case BACKGROUND_TASK_STATUS_RUNNABLE:
+		{
+			return CitusTaskStatusRunnableId();
+		}
+
+		case BACKGROUND_TASK_STATUS_RUNNING:
+		{
+			return CitusTaskStatusRunningId();
+		}
+
+		case BACKGROUND_TASK_STATUS_DONE:
+		{
+			return CitusTaskStatusDoneId();
+		}
+
+		case BACKGROUND_TASK_STATUS_ERROR:
+		{
+			return CitusTaskStatusErrorId();
+		}
+
+		case BACKGROUND_TASK_STATUS_UNSCHEDULED:
+		{
+			return CitusTaskStatusUnscheduledId();
+		}
+
+		case BACKGROUND_TASK_STATUS_CANCELLED:
+		{
+			return CitusTaskStatusCancelledId();
+		}
+	}
+
+	elog(ERROR, "unknown BackgroundTaskStatus");
+	return InvalidOid;
+}
+
+
+/*
+ * GetNextBackgroundJobsJobId reads and increments the SQL sequence associated with the
+ * background job's job_id. After incrementing the counter it returns the counter back to
+ * the caller.
+ *
+ * The return value is typically used to insert new jobs into the catalog.
+ */
+int64
+GetNextBackgroundJobsJobId(void)
+{
+	return DatumGetInt64(nextval_internal(DistBackgroundJobsJobIdSequenceId(), false));
+}
+
+
+/*
+ * GetNextBackgroundTaskTaskId reads and increments the SQL sequence associated with the
+ * background job tasks' task_id. After incrementing the counter it returns the counter
+ * back to the caller.
+ *
+ * The return value is typically used to insert new tasks into the catalog.
+ */
+int64
+GetNextBackgroundTaskTaskId(void)
+{
+	return DatumGetInt64(nextval_internal(DistBackgroundTasksTaskIdSequenceId(), false));
+}
+
+
+/*
+ * CreateBackgroundJob is a helper function to insert a new Background Job into Citus'
+ * catalog. After inserting the new job's metadataa into the catalog it returns the job_id
+ * assigned to the new job. This is typically used to associate new tasks with the newly
+ * created job.
+ */
+int64
+CreateBackgroundJob(const char *jobType, const char *description)
+{
+	Relation pgDistBackgroundJobs =
+		table_open(DistBackgroundJobsRelationId(), RowExclusiveLock);
+
+	/* insert new job */
+	Datum values[Natts_pg_dist_background_jobs] = { 0 };
+	bool nulls[Natts_pg_dist_background_jobs] = { 0 };
+
+	memset(nulls, true, sizeof(nulls));
```

Comment:
nit: why use `null = true` as the default? It seems using null is the exception, not the rule here.

Reply:
During development it helped when adding new fields to the table. Instead of assuming all fields are set we actually assume they are not set, until explicitly set.

When adding a new field we auto expand the arrays. If we add a new field and forget to add it here we would write random data into the table. Now we actually get an error (or assert - don't know exactly, expect an error) here. Which makes it easier to diagnose where we need to change something compared to chasing where random data was written.

Reply:
The nicer error message sounds useful indeed. Two things would help readability of the code though:
1. remove the newline right above these `memset` calls, to make them visually part of the initialization of the two arrays
2. Create a function similar to SetFieldValue (e.g. `InitFieldValue`) that does the same as SetFieldValue, but without the replace argument.

Reply:
I can imagine, although I actually wanted to add a 3rd function to set text values, which then adds 3 versions for Init.

How about we make replace optional in `SetFieldValue` ?